software/13.Rust/README.md

```diff
@@ -0,0 +1,165 @@
+
+#
+
+## Introduction to Rust
+
+### Step 1.1 - Hello World! in Rust
+
+> :exclamation: We strongly advise you to use the resources given for this exercise.
+
+To do this, create a file `main.rs` in a folder called `src`.
+
+Your main function should write `Hello World!` to your terminal.
+
+##### Resources
+
+ - [doc.Rust](https://doc.rust-lang.org/rust-by-example/hello.html)
+ - [Build Commands](https://doc.rust-lang.org/cargo/commands/build-commands.html)
+
+#
+
+### Step 1.2 - Palindrome ?
+
+> :exclamation: We strongly advise you to use the resources given for this exercise.
+
+The goal of the exercise is to create a function which takes as parameter a string `word` and returns true if it is a palindrome and false in the opposite case.
+
+##### Resources
+ - [What is a palindrome ?](https://www.wikiwand.com/en/Palindrome)
+ - [New function](https://doc.rust-lang.org/book/ch03-03-how-functions-work.html)
+ - [The `bool` in Rust](https://doc.rust-lang.org/std/primitive.bool.html)
+
+> :bulb: To test your functions easier, we advise you to take a look at the [`assert_eq`](https://doc.rust-lang.org/std/macro.assert_eq.html)
+
+#
+
+### Step 1.3 - Fibonacci number
+
+> :exclamation: We strongly advise you to use the resources given for this exercise.
+
+In this exercise we ask you to recreate a function able to take as parameters a number of starts and a maximum vector. From these data you must generate and display the Fibonacci number.
+
+Here is a small example of a part of the Fibonacci number:
+
+```shell
+0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, ...
+```
+
+##### Resources
+ - [Fibonacci number](https://www.wikiwand.com/en/Fibonacci_number)
+ - [Match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html)
+
+#
+
+### Step 1.4 - A simple and useful `Calculator` !
+
+> :exclamation: We strongly advise you to use the resources given for this exercise.
+
+A little bit harder !
+
+You will have to create a small calculator that will manage :
+```
+- addition
+- subtraction
+- multiplication
+- division
+```
+
+The numbers to manipulate must be recovered as you go along and must be entered by the user (Be careful with your error management).
+
+After your calculation you just have to display the result in your terminal.
+
+##### Resources
+ - [Module std::io](https://doc.rust-lang.org/std/io/index.html)
+ - [Examples io](https://doc.rust-lang.org/std/io/struct.Stdin.html#method.read_line)
+ - [Option & unwrap](https://doc.rust-lang.org/rust-by-example/error/option_unwrap.html)
+ - [Match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html)
+
+#
+
+### Step 1.5 - Pig Latin
+
+> :exclamation: We strongly advise you to use the resources given for this exercise.
+
+The serious stuff begins !
+
+The idea is inspired by `Pig Latin` but with a small variation nevertheless it can help you in learning a little more about the [original version](https://www.wikiwand.com/en/Pig_Latin).
```

Comment:
```suggestion
The idea is inspired by `Pig Latin` but with a small variation nevertheless it can help you  learning a little more about the [original version](https://www.wikiwand.com/en/Pig_Latin).
```

I don't understand the end ?